langchain4j-cloud-sql-postgres/src/test/java/dev/lanchain4j/engine/PostgresEngineIT.java

```
@@ -0,0 +1,227 @@
+                .storeMetadata(false)
+                .build();
+
+        engine.initVectorStoreTable(initParameters);
+
+        RuntimeException exception = assertThrows(RuntimeException.class, () -> {
+            engine.initVectorStoreTable(initParameters);
+        });
+
+        assertThat(exception.getMessage())
+                .isEqualTo(
+                        String.format("Failed to initialize vector store table: \"public\".\"%s\"", CUSTOM_TABLE_NAME));
+        // table name is truncated in PSQL exception
+        assertThat(exception.getCause().getMessage())
+                .isEqualTo(String.format(
+                        "ERROR: relation \"%s\" already exists",
+                        CUSTOM_TABLE_NAME.substring(0, CUSTOM_TABLE_NAME.length() - 2)));
+    }
+
+    @Test
+    void create_engine_with_iam_auth() throws SQLException {
+        PostgresEngine iamEngine = new PostgresEngine.Builder()
+                .projectId(projectId)
+                .region(region)
+                .instance(instance)
+                .database(database)
+                .user(user)
+                .password(password)
+                .ipType(ipType)
+                .iamAccountEmail(iamEmail)
+                .iamAccountEmail(iamEmail)
+                .build();
+        try (Connection connection = iamEngine.getConnection(); ) {
+            ResultSet rs = connection.createStatement().executeQuery("SELECT 1");
+            rs.next();
+            assertThat(rs.getInt(1)).isEqualTo(1);
+        }
+    }
+
+    @Test
+    void create_engine_with_get_iam_email() throws SQLException {
+        PostgresEngine iamEngine = new PostgresEngine.Builder()
+                .projectId(projectId)
+                .region(region)
+                .instance(instance)
+                .database(database)
+                .user(user)
+                .password(password)
+                .ipType(ipType)
+                .iamAccountEmail(iamEmail)
```

Do not set user if using IAM email. PLease throw an error if the user uses both user and iamaccountemail

---

now throwing error in engine  if both user and iamaccountemail are given- Only one of User or Iam account email is allowed